cdci_data_analysis/analysis/parameters.py

```diff
@@ -476,14 +469,11 @@ def __init__(self, value=None, T_format='isot', name=None, Time_format_name=None
 
         self._set_time(value, format=T_format)
 
-    def set_par(self, value, units=None):
-        # sets the value to the format initially specified in the form
-        if units is not None:
-            self.units = units
-        self.value = value
-        # return the relative isot format value
-        # afterwards also the units_name will be set to isot
-        return self._astropy_time.isot
+    def get_value_in_default_format(self, value) -> str:
```

Comment:
it does not always return str. Currently, it would be `Union[str, float, None]`. But the method should be modified to never return `None`, else unclear what happens to it.

Reply:
when do you see this returning `None`?

Reply:
https://github.com/oda-hub/dispatcher-app/blob/f0ee9e2bdb08084a754e2f6d7297b066b4727da2/cdci_data_analysis/analysis/parameters.py#L472-L477

there is no `else`. If neither of the two conditions is satisfied it returns `None`. This might happen if the validation elsewhere does not succeed, or allowed formats are redefined in inherited class, or something like that.

In any case, the method, as it is, may end up there, which will cause an unclear error.
If `else` can only happen in the case of some usage error, it should raise and exception.

Reply:
yes, indeed I reduced the method as you suggest in the other comment